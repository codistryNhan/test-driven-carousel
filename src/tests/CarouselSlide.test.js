import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

import CarouselSlide from '../CarouselSlide';

configure({ adapter: new Adapter() });

describe('CarouselSlide', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CarouselSlide
        imgUrl='https://example.com/image.png'
        description='A jaw droppingly spectacular image'
        attribution='Trevor Burnham'
      />
    );
  });

  it('renders a <figure>', () => {
    expect(wrapper.type()).toBe('figure');
  });

  it('renders an <img> and a <figcaption> as children', () => {
    expect(wrapper.childAt(0).type()).toBe('img');
    expect(wrapper.childAt(1).type()).toBe('figcaption');
  });

  it('passes `imgUrl` through to the <img>', () => {
    const imgUrl = 'https://example.com/image.png';
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(imgUrl);
  });

  it('uses `description` and `attribution` as the <figcaption>', () => {
    const description = 'A jaw droppingly spectacular image';
    const attribution = 'Trevor Burnham';

    expect(wrapper.find('figcaption').text()).toBe(
      `${description} ${attribution}`
    );
    expect(wrapper.find('figcaption strong').text()).toBe(description);
  });

  it('passes other props to the <figure>', () => {
    const style = {};
    const onClick = () => {};
    const className = 'my-carousel-slide';

    wrapper.setProps({ style, onClick, className });
    expect(wrapper.prop('style')).toBe(style);
    expect(wrapper.prop('onClick')).toBe(onClick);
    expect(wrapper.prop('className')).toBe(className);
  });
});
