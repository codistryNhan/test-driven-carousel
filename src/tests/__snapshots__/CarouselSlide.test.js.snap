// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlide renders correctly 1`] = `
<figure
  Img={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CarouselSlide__DefaultImg-ds4ujd-0",
        "isStatic": false,
        "rules": Array [
          "object-fit:cover;width:100%;height:",
          [Function],
          ";",
        ],
      },
      "displayName": "CarouselSlide__DefaultImg",
      "render": [Function],
      "styledComponentId": "CarouselSlide__DefaultImg-ds4ujd-0",
      "target": "img",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
>
  <CarouselSlide__DefaultImg
    imgHeight={500}
    src="https://example.com/image.png"
  />
  <figcaption>
    <strong>
      Description
    </strong>
     
    Attribution
  </figcaption>
</figure>
`;
